src/main/java/seedu/address/logic/parser/DeleteInfoCommandParser.java

```diff
@@ -0,0 +1,97 @@
+
+import java.util.Arrays;
+
+import seedu.address.logic.commands.DeleteInfoCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Nric;
+
+
+
+
+
+/**
+ * Parses input arguments and creates a new DeleteInfoCommand object
+ */
+public class DeleteInfoCommandParser implements Parser<DeleteInfoCommand> {
+    /**
+     * List of prefixes for optional fields, in the order of the fields in DeleteInfoCommand.
+     */
+    public static final Prefix[] OPTIONAL_PREFIXES = {PREFIX_EMAIL, PREFIX_ALLERGIES, PREFIX_BLOODTYPE,
+        PREFIX_DATEOFADMISSION, PREFIX_COUNTRY, PREFIX_CONDITION, PREFIX_SYMPTOM, PREFIX_DIAGNOSIS};
+    /**
+     * Parses the given {@code String} of arguments in the context of the DeleteInfoCommand
+     * and returns a DeleteInfoCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteInfoCommand parse(String args) throws ParseException {
+        //Check for empty input
+        if (args.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteInfoCommand.MESSAGE_USAGE));
+        }
+        //Check for NRIC
+        Nric nric;
+        String[] splitedArgs = args.trim().split("\\s+");
+        try {
+            nric = ParserUtil.parseNric(splitedArgs[0]);
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteInfoCommand.MESSAGE_USAGE), pe);
+        }
+        //Parse prefixes
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NRIC, PREFIX_NAME, PREFIX_PHONE, PREFIX_ADDRESS,
+                        PREFIX_DATEOFBIRTH, PREFIX_SEX, PREFIX_STATUS, PREFIX_TAG, PREFIX_EMAIL, PREFIX_COUNTRY,
+                        PREFIX_DATEOFADMISSION, PREFIX_ALLERGIES, PREFIX_BLOODTYPE, PREFIX_CONDITION, PREFIX_SYMPTOM,
+                        PREFIX_DIAGNOSIS);
+        //Check for empty input
+        if (argMultimap.length() <= 1) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteInfoCommand.MESSAGE_USAGE));
+        }
+        //check for mandatory fields
+        Prefix[] mandatoryPrefixes = {PREFIX_NRIC, PREFIX_NAME, PREFIX_PHONE, PREFIX_ADDRESS, PREFIX_DATEOFBIRTH,
+            PREFIX_SEX, PREFIX_STATUS};
+        if (Arrays.stream(mandatoryPrefixes).anyMatch(argMultimap::contains)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteInfoCommand.MESSAGE_ONLY_OPTIONAL_FIELDS));
+        }
+
+        //Check for optional fields & any extra invalid input
+        boolean[] fieldsToDelete = new boolean[DeleteInfoCommand.NUM_FIELDS];
+        Arrays.fill(fieldsToDelete, false);
+        Prefix[] optionalPrefixes = {PREFIX_EMAIL, PREFIX_ALLERGIES,
```

Comment:
Might be better to reuse the optional prefix array that was already defined instead of creating a duplicate